php/class-customize-snapshot-manager.php

```diff
@@ -123,6 +123,11 @@ public function __construct( Plugin $plugin ) {
 		add_action( 'customize_save_after', array( $this, 'save_snapshot' ) );
 		add_action( 'admin_bar_menu', array( $this, 'customize_menu' ), 41 );
 		add_action( 'customize_controls_print_footer_scripts', array( $this, 'render_templates' ) );
+		
+		add_filter( 'theme_mod_nav_menu_locations', array( $this, 'filter_theme_mod_nav_menu_locations' ) );
+		add_filter( 'wp_get_nav_menus', array( $this, 'filter_wp_get_nav_menus' ) );
+		add_filter( 'wp_get_nav_menu_items', array( $this, 'filter_wp_get_nav_menu_items' ), 10, 3 );
+		add_filter( 'wp_get_nav_menu_object', array( $this, 'filter_wp_get_nav_menu_object' ), 10, 2 );
```

Comment:
Actually while debugging I saw that these filters are already used via `Customize_Snapshot_Manager::preview()`.
https://github.com/xwp/wp-customize-snapshots/blob/master/php/class-customize-snapshot-manager.php#L862-L887

If the `$setting` is a menu setting, the filters are set by `WP_Customize_Nav_Menu_Setting::preview()` but they are not making a difference right now for the Menus section. They do work in case of widgets, e.g. the menus created via snapshot are displayed in the selection list when adding a `Custom Menu` widget.

I'm right now testing the filters used in `WP_Customize_Nav_Menus` as suggested to see if there's still something missing.